### PR TITLE
Mapi 1194 de duplicate

### PR DESCRIPTION
**Pull request by alexduf**

### CODE REVIEW

**Comment by alexduf** [comment]
Any volunteer for this one?

File: registration/app/registration/services/WindowsNotificationRegistrar.scala [hunk 1]
```diff
@@ -22,11 +22,10 @@ class WindowsNotificationRegistrar(hubClient: NotificationHubClient)(implicit ec
       .update(regId, RawWindowsRegistration.fromMobileRegistration(registration))
       .map(hubResultToRegistrationResponse)
 
-    val channelUri = registration.deviceId
-    hubClient.registrationsByChannelUri(channelUri).flatMap {
+    hubClient.registrationsByChannelUri(oldChannelUri).flatMap {
```
Comment (agulowaty) [inline registration/app/registration/services/WindowsNotificationRegistrar.scala]: shouldn't we also check if channelUri from Registration (deviceId) is already taken by some registration in hub ? Or we can optimistically assume device will not send us existing channel in new Registration

File: registration/app/registration/services/WindowsNotificationRegistrar.scala [hunk 2]
```diff
@@ -13,7 +13,15 @@ import scalaz.std.option.optionSyntax._
...
-  override def register(registration: Registration): Future[\/[Error, RegistrationResponse]] = {
+  override def register(lastKnownChannelUri: String, registration: Registration): Future[\/[Error, RegistrationResponse]] = {
+    hubClient.registrationsByChannelUri(registration.deviceId).flatMap {
+      case \/-(Nil) => createOrUpdateRegistration(lastKnownChannelUri, registration)
+      case \/-(_ :: _) => createOrUpdateRegistration(registration.deviceId, registration)
+      case -\/(e: Error) => Future.successful(e.left)
+    }
+  }
+
+  def createOrUpdateRegistration(lastKnownChannelUri: String, registration: Registration): Future[\/[Error, RegistrationResponse]] = {
```
Comment (agulowaty) [inline registration/app/registration/services/WindowsNotificationRegistrar.scala]: [minor] should be private

**Comment by agulowaty** [comment]
:+1:

**Pull request merged**